src/routes/breaddit/breaddit.controller.js

```diff
@@ -0,0 +1,80 @@
+const fs = require("fs");
+const { promisify } = require("util");
+const writeFile = promisify(fs.writeFile);
+const { validationResult } = require("express-validator");
+
+const breadditdata = require("../../../db/breaddit.data.json");
+
+const breadditController = (req, res) => {
+  return res.json({
+    data: breadditdata
+  });
+};
+
+const getBreadditById = (req, res) => {
+  const breadditId = req.params.id;
+
+  const posts = breadditdata.find(post => post.id === breadditId);
+
+  if (posts) {
+    return res.json(posts);
+  } else {
+    res.status(404).send("Breaddit not found!");
+  }
+};
+const updateBreaddit = (req, res) => {
+  const breadditId = req.params.id;
+  const breaddit = req.body;
+  const posts = breadditdata.find(post => post.id === breadditId);
+
+  const filteredPosts = breadditdata.filter(post => post.id !== breadditId);
+  const updatedPosts = [...filteredPosts, breaddit];
+
+  if (posts) {
+    return res.json(updatedPosts);
+  } else {
+    res.status(404).send("Breaddit not found!");
+  }
+};
+
+const deleteBreaddit = async (req, res) => {
+  const breadditId = req.params.id;
+  const filteredPosts = breadditdata.filter(post => post.id !== breadditId);
+  const posts = breadditdata.find(post => post.id === breadditId);
+
+  if (posts) {
+    return res.json(filteredPosts);
+  } else {
+    res.status(404).send("Breaddit not found!");
+  }
+};
+
+const postBreaddit = async (req, res) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() });
+  }
+  const id = breadditdata.length + 1;
+
+  const postBreadditData = {
+    id,
+    ...req.body
+  };
+  await writeFile(
+    "db/breaddit.data.json",
+    JSON.stringify([...breadditdata, postBreadditData])
+  );
+  res.status(201);
+  return res.json({
+    id,
+    ...req.body
+  });
+};
+
+module.exports = {
+  breadditController,
+  postBreaddit,
+  getBreadditById,
+  updateBreaddit,
+  deleteBreaddit
+};
```

Comment:
More of a nitpick: `post` may have been a more descriptive variable name than `breaddit`, seeing as you're manipulating posts in this controller, and not the entire breaddit system.